File: Firebase/creative_api/functions/src/Repository/VehicleRepository.js
const db = require("./../Config/ConnectionDB");
const vehicleRef = db.collection("vehicles");

const Vehicle = require("./../Class/Vehicle")

class VehicleRepository {
    async save(vehicle) {
        return (await vehicleRef.add(vehicle).then((doc) => {
            vehicle.id = doc.id
            return new Vehicle(vehicle);
        }).catch(() => {
            return null;
        }));
    }

    async update(vehicle) {
        let id = vehicle.id;
        delete vehicle.id;
        return (await vehicleRef.doc(id).update(vehicle).then(() => {
            return "ok";
        }).catch(() => {
            return null;
        }));
    }
    async find(id) {
        return (await vehicleRef.doc(id).get()
            .then((querySnapshot => {
                let vehicle = {};
                vehicle = querySnapshot.data();
                vehicle.id = querySnapshot.id;
                return new Vehicle(vehicle);
            })).catch((e) => {
                console.log(e);
                return null;
            }));
    }
    async findAll() {
        return (await vehicleRef.get().then((querySnapshot => {
                var vehicles = [];
                querySnapshot.docs.forEach((doc) => {
                    let vehicleDoc = doc.data();
                    vehicleDoc.id = doc.id;
                    vehicles.push(new Vehicle(vehicleDoc).get());
                });
                return vehicles;
            })).catch(() => {
                return null;
            }));
    }

    async findByLicensePlate(license_plate) {
        return (await vehicleRef.where("license_plate", "==", license_plate).get().then((querySnapshot => {
            console.log(querySnapshot.docs.length);
            if (querySnapshot.docs.length >= 1) {
                return null;
            }
            return new Vehicle({});
        })).catch(() => {
            return null;
        }));
    }
}

module.exports = new VehicleRepository();

File: classes PI/Truckleer/Truckleer/Modules/Vehicle/VehicleRepository.cs
using Google.Cloud.Firestore;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Truckleer.Modules
{
    class VehicleRepository
    {
        //Collection reference property
        readonly CollectionReference Reference;
        //Constructor Class
        public VehicleRepository()
        {
            //Create firestore connection
            ConnectionFirestore coon = new ConnectionFirestore();
            //Initializate Reference
            Reference = coon.Db.Collection("vehicles");
        }

        //Method for get All vehicles
        async public Task<List<Vehicle>> FindAll()
        {
            //Create a list of vehicles
            List<Vehicle> vehicles = new List<Vehicle>();
            //Create a Query without any filter
            QuerySnapshot snapshot = await Reference.GetSnapshotAsync();

            //Pass all Documents
            foreach (DocumentSnapshot queryResult in snapshot.Documents)
            {
                //Convert Document in a Vehicle class
                Vehicle us = queryResult.ConvertTo<Vehicle>();
                //Set id of vehicle
                us.Id = queryResult.Id;
                //Add vehicle to list
                vehicles.Add(us);
            }
            //Return List
            return vehicles;
        }
        //Method for find One vehicle by Id
        async public Task<Vehicle> Find(string Id)
        {
            //Create a Document Reference with id like /vehicles/id_value
            DocumentSnapshot DocRef = await Reference.Document(Id).GetSnapshotAsync();
            //Initializate a vehicle  with null value
            Vehicle us = null;
            //Check if document exists
            if (DocRef.Exists)
            {
                //Convert document to a Vehicle class
                us = DocRef.ConvertTo<Vehicle>();
                //Set id of vehicle
                us.Id = DocRef.Id;
            }
            //Return vehicle
            return us;
        }
        //Method for save or update Vehicle
        async public Task<bool> Save(Vehicle vehicle)
        {
            //Check if vehicle exisit
            if (vehicle.Id == null)//If not exist
            {
                //Create new vehicle
                DocumentReference snapshot = await Reference.AddAsync(vehicle);
                //return a bool if is successful
                return snapshot.Id != null;
            }
            else
            {
                //update vehicle and merge values
                WriteResult snapshot = await Reference.Document(vehicle.Id).SetAsync(vehicle, SetOptions.MergeAll);
                //return a bool if is successful
                return snapshot.UpdateTime != null;
            }
        }
    }
}
